openms-wiki/developer-faq.md

```diff
@@ -0,0 +1,490 @@
+Developer FAQ
+=============
+
+The following contains answers to typical questions from developers about OpenMS.
+
+> **_NOTE:_**  Some actions require special permissions like e.g. updating the website.
+
+## General
+
+The following section provides general information to new contributors.
+
+### I am new to OpenMS. What should I do first?
+
+* Check out the development version of OpenMS (see website).
+* Try and build OpenMS according to the installation instructions.
+* Read the OpenMS Coding Convention.
+* Read the OpenMS Tutorial.
+* Create a GitHub account
+* Register to the open-ms-general and open-ms-developers mailing list. (You can see the developers list only if you are logged in to Sourceforge and if you are a OpenMS developer).
+
+### What is the difference between an OpenMS tool and util?
+
+A tool starts its lifecycle in `UTILS` and may exist without being thoroughly tested. Tools may be promoted from `UTILS` to `TOOLS` if they are stable enough, are fully tested, fully documented, and a test workflow exists.
+
+### I have written a class for OpenMS. What should I do?
+
+Follow the [OpenMS coding conventions](coding-conventions.md).
+
+Coding style (brackets, variable names, etc.) must conform to the conventions.
+
+* The class and all the members must be documented thoroughly.
+* You can check your code with the tool  `tools/checker.php`. Call `php tools/checker.php` for detailed instructions.
+
+Please open a pull request and follow the [checklist](pull-request-checklist.md).
+
+### How do I update the [openms.de](https://www.openms.de)?
+
+Login to the [wordpress admin area](www.openms.de/wp-admin) with your username and password assigned by the current Homepage maintainers.
```

Comment:
re-write the answer for an external contributor.

Reply:
How do you update the website if you are an external contributor? Should I just delete this question?

Reply:
yes, delete from developer-faq.md.